src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/email/provide/UCCollectionProvideEmails.java

```diff
@@ -18,15 +19,16 @@ public record UCCollectionProvideEmails(
     ProvideAllUnreadEmailsService provideAllUnreadEmailsService,
     ProvideAllEmailsService provideAllEmailsService) {
 
-  public static UCCollectionProvideEmails init(FileMailboxRepository mailboxRepository) {
+  public static UCCollectionProvideEmails init(
+      AuthSessionUseCase authSession, FileMailboxRepository mailboxRepository) {
     return new UCCollectionProvideEmails(
-        new ProvideDeletedEmailsService(mailboxRepository),
-        new ProvideSpamEmailsService(mailboxRepository),
-        new ProvideInboxReadEmailsService(mailboxRepository),
-        new ProvideInboxUnreadEmailsService(mailboxRepository),
-        new ProvideSentEmailsService(mailboxRepository),
-        new ProvideAllReadEmailsService(mailboxRepository),
-        new ProvideAllUnreadEmailsService(mailboxRepository),
-        new ProvideAllEmailsService(mailboxRepository));
+        new ProvideDeletedEmailsService(authSession, mailboxRepository),
```

Comment:
the authsession is there to determine the current user right ? but how did we do it before without it ?

Reply:
okay, I got it, because we entered the authsessionuser in the method

Reply:
Correct! Previously we basicallly had the AuthSession in the "frontend" / CLI. Now the services got the AuthSession to get the current UserAddress